components/support/nimbus-cli/src/value_utils.rs

```diff
@@ -159,6 +165,63 @@ pub(crate) fn try_find_features(value: &Value) -> Result<Vec<Value>> {
     })
 }
 
+pub(crate) fn try_find_mut_features_from_branch<'a>(
+    value: &'a mut Value,
+) -> Result<HashMap<String, &'a mut Value>> {
+    let mut res = HashMap::new();
+    if value.has("features") {
+        let features = value.get_mut_array("features")?;
+        for f in features {
+            res.insert(
+                f.get_str("featureId")?.to_string(),
+                f.get_mut_object("value")?,
+            );
+        }
+    } else {
+        let f: &'a mut Value = value.get_mut_object("feature")?;
+        res.insert(
+            f.get_str("featureId")?.to_string(),
+            f.get_mut_object("value")?,
+        );
+    }
+    Ok(res)
+}
+
+pub(crate) fn patch_value(this: &mut Value, patch: &Value) -> bool {
+    match (this, patch) {
+        (Value::Object(t), Value::Object(p)) => patch_map(t, p),
+        (Value::String(t), Value::String(p)) => *t = p.clone(),
+        (Value::Bool(t), Value::Bool(p)) => *t = *p,
+        (Value::Number(t), Value::Number(p)) => *t = p.clone(),
+        (Value::Array(t), Value::Array(p)) => *t = p.clone(),
+        (Value::Null, Value::Null) => (),
+        _ => return false,
+    };
+    true
+}
+
+fn patch_map(this: &mut Map<String, Value>, patch: &Map<String, Value>) {
+    for (k, v) in patch {
+        match (this.get_mut(k), v) {
+            (Some(_), Value::Null) => {
+                this.remove(k);
+            }
+            (_, Value::Null) => {
+                // If the patch is null, then don't add it to this value.
+            }
+            (Some(t), p) => {
+                if !patch_value(t, p) {
+                    println!("Warning: the patched key '{k}' has different types: {t} != {p}");
+                    this.insert(k.clone(), v.clone());
+                }
+            }
+            (None, _) => {
+                this.insert(k.clone(), v.clone());
+            }
+        }
+    }
+}
```

Comment:
Yes, you're absolutely right. I thought I'd take a short cut– since there was only going to one implementation of the trait, but there ended up being two.